service/src/main/java/uk/nhs/adaptors/gpc/consumer/filters/ReactorMdcSupport.java

```diff
@@ -0,0 +1,95 @@
+package uk.nhs.adaptors.gpc.consumer.filters;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+
+import org.reactivestreams.Publisher;
+import org.reactivestreams.Subscription;
+import org.slf4j.MDC;
+import org.springframework.context.annotation.Configuration;
+
+import reactor.core.CoreSubscriber;
+import reactor.core.Fuseable;
+import reactor.core.publisher.Hooks;
+import reactor.core.publisher.Operators;
+import reactor.util.context.Context;
+
+@Configuration
+public class ReactorMdcSupport {
+
+    private static final String MDC_HOOK_KEY = "MDC";
+    private static final Function<? super Publisher<Object>, ? extends Publisher<Object>> LIFTER =
+        Operators.liftPublisher(
+            publisher -> {
+                if (MDC.getCopyOfContextMap() == null) {
+                    return false;
+                }
+                return !(publisher instanceof Fuseable.ScalarCallable);
+            },
```

Comment:
Possible alternative
```suggestion
            publisher -> MDC.getCopyOfContextMap() != null
                && !(publisher instanceof Fuseable.ScalarCallable),
```